spec/datadog/appsec/contrib/rails/integration_test_spec.rb

```diff
@@ -0,0 +1,284 @@
+# typed: ignore
+
+require 'datadog/tracing/contrib/rails/rails_helper'
+
+require 'datadog/appsec'
+
+RSpec.describe 'Rails integration tests' do
+  include Rack::Test::Methods
+
+  let(:sorted_spans) do
+    chain = lambda do |start|
+      loop.with_object([start]) do |_, o|
+        # root reached (default)
+        break o if o.last.parent_id == 0
+
+        parent = spans.find { |span| span.span_id == o.last.parent_id }
+
+        # root reached (distributed tracing)
+        break o if parent.nil?
+
+        o << parent
+      end
+    end
+    sort = ->(list) { list.sort_by { |e| chain.call(e).count } }
+    sort.call(spans)
```

Comment:
Slightly confused -- this sort lambda doesn't seem to be reused anywhere else. Is this just a really roundabout way of just `spans.sort_by { |e| chain.call(e).count }`...?

Reply:
I think I should extract that away, I used it a lot in`spec/datadog/tracing/contrib/sinatra/tracer_spec.rb` and copied it over (BAD!)